video_xblock/video_xblock.py

```diff
@@ -28,7 +31,48 @@
 log = logging.getLogger(__name__)
 
 
-class VideoXBlock(StudioEditableXBlockMixin, XBlock):
+class TranscriptsMixin(XBlock):
+    """
+    TranscriptsMixin class to encapsulate transcripts-related logic
+    """
+
+    @staticmethod
+    def convert_caps_to_vtt(caps):
+        """
+        Utility method conoverts any supported transcripts into WebVTT format.
+        Supported input formats: DFXP/TTML - SAMI - SCC - SRT - WebVTT
+        """
+
+        reader = detect_format(caps)
+        if reader:
+            return WebVTTWriter().write(reader().read(caps))
```

Comment:
Nope. Since we're looking for a client-side solution anyway.
Caching will overcomplicate current solution.